site/source/docs/porting/asyncify.rst

```diff
@@ -217,10 +238,46 @@ You can build this with
 This example calls the Promise-returning ``window.crypto.subtle()`` API (the
 example is based off of
 `this MDN example <https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest#Basic_example>`_
-). Note how we pass the value to be returned into ``wakeUp()``. We must also
-return the value returned from ``handleSleep()``. The calling C code then
+).
+
+Note that we must propagate the value returned from ``handleSleep()``. The calling C code then
 gets it normally, after the Promise completes.
 
+If you're using ``handleSleep`` API, the value needs to be also passed to the ``wakeUp`` callback, instead of being returned from our handler:
+
+.. code-block:: cpp
+
+    // ...
+    return Asyncify.handleSleep(wakeUp => {
+      const text = UTF8ToString(str);
+      const encoder = new TextEncoder();
+      const data = encoder.encode(text);
+      out("ask for digest for " + text);
+      window.crypto.subtle.digest("SHA-256", data).then(digestValue => {
+        out("got digest of length " + digestValue.byteLength);
+        // Return the value by sending it to wakeUp(). It will then be returned
+        // from handleSleep() on the outside.
+        wakeUp(digestValue.byteLength);
+      });
+    });
+    // ...
+
+Usage with Embind
+#################
+
+If you're using :ref:`Embind<embind-val-guide>` for interaction with JavaScript
+and want to ``await`` a dynamically retrieved ``Promise``, you can call an
+``await()`` method directly on the ``val`` instance:
+
+.. code-block:: cpp
+
+    val my_object = /* ... */;
+    val result = my_object.call("someAsyncMethod").await();
+
+In this case you don't need to worry about ``ASYNCIFY_IMPORTS``, since it's an
+internal implementation detail of ``val::await`` and Emscripten takes care of it
+automatically.
```

Comment:
wait, how does that work under the hood?

Reply:
I've added `_emval_await` to default `ASYNCIFY_IMPORTS` list.